packages/parser/src/actions/ZoneChange.ts

```diff
@@ -0,0 +1,31 @@
+import { ILogLine } from '../types';
+
+export interface ZoneChangeInfo {
+  timestamp: number;
+  instanceId: number;
+  zoneName: string;
+  difficultyID: number;
+}
+
+export class ZoneChange implements ZoneChangeInfo {
+  public static supports(logLine: ILogLine): boolean {
+    return logLine.event.startsWith('ZONE_CHANGE');
+  }
+
+  public readonly timestamp: number;
+  public readonly instanceId: number;
+  public readonly zoneName: string;
+  public readonly difficultyID: number;
+
+  constructor(public readonly logLine: ILogLine) {
+    if (!ZoneChange.supports(logLine)) {
+      throw new Error('Event not supported as ZoneChange: ' + logLine.raw);
+    }
+
+    this.timestamp = logLine.timestamp;
+
+    this.instanceId = logLine.parameters[0];
```

Comment:
is this an ID of the zone or unique ID of the arena instance? If it's zone ID, it might be better to use this instead of zone name in downstream logic because zone name can be different when people play from clients with different languages.